security/pkg/nodeagent/sds/sdsservice_test.go

```diff
@@ -492,6 +516,157 @@ func TestStreamSecretsMultiplePush(t *testing.T) {
 		t.Fatalf("failed to send push notificiation to proxy %q", conID)
 	}
 	notifyChan <- "receive secret"
+	// Wait for pushSDS to return.
+	time.Sleep(2 * time.Second)
```

Comment:
is it possible to poll for this? sleep is not reliable and makes our unit tests slow